sdk/py/apepay/__init__.py

```diff
@@ -248,23 +248,24 @@ def streams_by_creator(self, creator: AddressType) -> Iterator["Stream"]:
         for stream_id in range(self.contract.num_streams(creator)):
             yield Stream(manager=self, creator=creator, stream_id=stream_id)
 
-    def all_streams(self) -> Iterator["Stream"]:
+    def all_streams(self, start_block: Optional[int] = None) -> Iterator["Stream"]:
         for stream_created_event in self.contract.StreamCreated.range(
-            self.contract.receipt.block_number, self.chain_manager.blocks.head.number
+            start_block or self.contract.receipt.block_number,
```

Comment:
Instead of `start_block or self.contract.receipt.block_number` you may want `start_block if start_block is not None else  self.contract.receipt.block_number` to allow the value of `0` for testnets.